### PR TITLE
Implemented pop up and serch category.

### PR DESCRIPTION
**Pull request by nag243**
----

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -14,7 +16,7 @@ import { TopbarComponent } from './shared/shared/components/topbar/topbar.compon
   imports: [
     BrowserModule,
     AppRoutingModule,
-    SharedModule,
+    SharedModule,NgbModule,NgbAlertModule
```
Comment (Nithyam1330) [inline src/app/app.module.ts]: Remove ngb modules

File: src/app/views/admin-views/category-list/category-list.component.html [hunk 2]
```diff
@@ -1,29 +1,34 @@
...
+            <thead>
+                <tr>
+                    <th scope="col">Id</th>
+                    <th scope="col">Category</th>
+                    <th scope="col">Created Date</th>
+                    <th scope="col">Updated Date</th>
+                </tr>
+            </thead>
+            <tbody>
+                <tr *ngFor="let product of products as i">
```
Comment (Nithyam1330) [inline src/app/views/admin-views/category-list/category-list.component.html]: As I should be changed, Change products to category

File: src/app/views/admin-views/category-list/category-list.component.ts [hunk 3]
```diff
@@ -1,29 +1,35 @@
...
-      id: '002',
-      catagory:'general',
-      currentdate:'19022020',
-      updateddate: '25012020'
-    },
-  ];
-
-  constructor() { }
-
+  products: any[] = [
```
Comment (Nithyam1330) [inline src/app/views/admin-views/category-list/category-list.component.ts]: Should be categories

File: src/app/views/admin-views/category-list/category-list.module.ts [hunk 4]
```diff
@@ -1,18 +1,25 @@
...
 @NgModule({
-  declarations: [CategoryListComponent],
+  declarations: [CategoryListComponent,CategoryComponent
+],
   imports: [
     CommonModule,
     CategoryListRoutingModule,
-    SharedModule,
-  ]
+    SharedModule,NgbModule
```
Comment (Nithyam1330) [inline src/app/views/admin-views/category-list/category-list.module.ts]: Remove Ngb Module

File: src/app/views/admin-views/category/category.component.ts [hunk 5]
```diff
@@ -1,4 +1,6 @@
 import { Component, OnInit } from '@angular/core';
+import { NgbModalRef, NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
```
Comment (Nithyam1330) [inline src/app/views/admin-views/category/category.component.ts]: Remove Unused imports

File: src/app/views/admin-views/category/category.component.ts [hunk 6]
```diff
@@ -7,7 +9,7 @@ import { Component, OnInit } from '@angular/core';
 })
 export class CategoryComponent implements OnInit {
 
-  constructor() { }
+  constructor( private modal: NgbActiveModal) { }
```
Comment (Nithyam1330) [inline src/app/views/admin-views/category/category.component.ts]: Remove unwanted service integration

File: src/app/app.module.ts (hunk 1 above)
Comment (nag243) [reply src/app/app.module.ts]: Resolved.

File: src/app/views/admin-views/category-list/category-list.component.html (hunk 2 above)
Comment (nag243) [reply src/app/views/admin-views/category-list/category-list.component.html]: Resolved.

File: src/app/views/admin-views/category-list/category-list.component.ts (hunk 3 above)
Comment (nag243) [reply src/app/views/admin-views/category-list/category-list.component.ts]: Changes as categories....Resolved.

File: src/app/views/admin-views/category-list/category-list.module.ts (hunk 4 above)
Comment (nag243) [reply src/app/views/admin-views/category-list/category-list.module.ts]: Removed.

File: src/app/views/admin-views/category/category.component.ts (hunk 5 above)
Comment (nag243) [reply src/app/views/admin-views/category/category.component.ts]: Resolved.

File: src/app/views/admin-views/category/category.component.ts (hunk 6 above)
Comment (nag243) [reply src/app/views/admin-views/category/category.component.ts]: NgbActiveModal
 this is for cancle the pop up,i tested,its not working without ngbactiveModal.

File: src/app/views/admin-views/category-list/category-list.module.ts [hunk 7]
```diff
@@ -1,18 +1,25 @@
 import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
+import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
```
Comment (Nithyam1330) [inline src/app/views/admin-views/category-list/category-list.module.ts]: Remove this

**Pull request merged**